### ISSUE TITLE
 "Enable Feature by JWT" plugin is active on the server:-not working

### ISSUE THREAD

**Issue by ReshmaJaiswal** [H2-invent/jitsi-admin#529]
after checking this and on server side when i am joing meet , it doesnt ask for token nor it sends jwt token via url ,its working normal only with or without checking checkbox.Is there any other way to enable it so that token is generated ???/

**Comment by ReshmaJaiswal** [comment]
what is appid and app secret ?? do we need to add this for jwt token generation??? is it from server side or keycloak id and secret? pls explain if i need to enable jwt for my server then what are requirements???

**Comment by holema** [comment]
Yes you need to set App Id and App Secret. This generates the JWT for the jitsi server.
https://github.com/jitsi/lib-jitsi-meet/blob/master/doc/tokens.md
https://github.com/nvonahsen/jitsi-token-moderation-plugin

You have to follow these instructions.

**Comment by ReshmaJaiswal** [comment]
i want to activate jwt for my server

**Comment by holema** [comment]
Then you have to install the jwt plugin in the jitsi server. You will get the app if and app secret after installing the plugin.

**Comment by ReshmaJaiswal** [comment]
can we refined JWT payload structure,if yes how ?

**Comment by holema** [comment]
you can only choose where the moderator claim should be located but not the whole JSON structure. why do you want to do this?

**Comment by ReshmaJaiswal** [comment]
when i am changing payload structure everyone joins as a moderator  , in this code  public function generateJwtPayload($userName, Rooms $room, Server $server, $moderator, RoomsUser $roomUser = null, $avatar = null)
    {
        if (!$server->getAppId()) {
            return null;
        }
        $payload = [
            "context" => [
                "user" => [
                    "name" => $userName,
                ],
            ],
            "moderator" => $moderator,
            "aud" => "jitsi_admin",
            "iss" => "jitsi_admin",
            "sub" => "*",
            "room" => "*",
            "exp" => time() + 3600, // Adjust expiration time as needed
        ];

        if ($roomUser && !$avatar) {
            if ($roomUser->getUser() && $roomUser->getUser()->getProfilePicture()) {
                $avatar = $this->uploadHelper->asset($roomUser->getUser()->getProfilePicture(), 'documentFile');
            }
        }
        if ($avatar) {
            $payload['context']['user']['avatar'] = $avatar;
        }
        return $payload;
    }   and what should i set in adding server option , Moderator position in JWT token ?? root claim or user claim?

**Comment by holema** [comment]
Hello @ReshmaJaiswal It seems to me that you made a fork of the repo where you make your changes. You have to make you fork with all changes Open Source ander the AGPL license too.  SO please send me a link where I can seee your changes so I can answer your question. I don't understand in you comment what you want to do.

**Comment by ReshmaJaiswal** [comment]
i want to just ask that in my jwtpayload we have parameter moderator in which it goes to backend server true or false when jwt enable , but in my backend server everyone joining as moderator, y such behaivour??? and in entry link also jwt is going along with joining url when enable jwt ?

**Comment by ReshmaJaiswal** [comment]
and what if i want to increse uid length i am changing in i am changing in roomcontroller.php but its not reflecting $roomNew->setUid(rand(01, 99) . time()); ?? i want to increse length of uid to 8 digit

**Comment by holema** [comment]
@ReshmaJaiswal The moderator propertiy is set in the jwt claim root or user, which can be selected in the server settings in the GUI. If the moderator token is not correctly used from you jitsi server, then I would highly suggest to double check your jitsi server, because there must be the error. IN all our installations it is working so it must be your jitsi server which ist not correctly reading the JWT.

For the other question the uid is coupled to the unix time.

**Comment by ReshmaJaiswal** [comment]
propertiy is set in the jwt claim root or user ? whats the difference between both??? root claim and user claim??

**Comment by ReshmaJaiswal** [comment]
is token appended to entry link also ?? what if i want entry link without token?

**Comment by holema** [comment]
then leave app id and app secret empty

**Comment by holema** [comment]
in the server setting in the GUI, where you set up the server url and server name

**Comment by ReshmaJaiswal** [comment]
then f i leave app id and app secret empty then for other url jwt will work??

**Comment by ReshmaJaiswal** [comment]
i observe the unexpected behaivour when i enable jwt then everyone joing as a participant and via entry link all became moderator ,means everyone receives token?what if i remove app id and secret what will be result?
pls help or i amm using application wrong way?
